### PR TITLE
#19275 Fixing

### PR DESCRIPTION
**Pull request by freddyucv**
Fixing https://github.com/dotCMS/core/issues/19275#issuecomment-742616200

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/authentication/ApiTokenResource.java [hunk 1]
```diff
@@ -166,19 +160,27 @@ public final Response deleteApiToken(@Context final HttpServletRequest request,
...
-        final User forUser = Try.of(()->APILocator.getUserAPI().loadUserById(formData.userId,requestingUser, false )).getOrNull();
-        
-        if(forUser == null) {
+        final User forUser = formData.userId != null ? getUserById(formData, requestingUser) : requestingUser;
+
+        if (requestingUser != forUser && !requestingUser.isAdmin()) {
+            throw new DotDataException("Just Admin user can request a Token for another user");
+        }
 
+        if(forUser == null) {
```
Comment (victoralfaro-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/authentication/ApiTokenResource.java:176]: Shouldn't this be the first evaluation?

Comment (freddyucv) [reply dotCMS/src/main/java/com/dotcms/rest/api/v1/authentication/ApiTokenResource.java:172]: done https://github.com/dotCMS/core/pull/19698/commits/95a2812fa9c6ac8f3ec611d7223a4716794ad726

**Review by victoralfaro-dotcms — APPROVED** [review]
Looks good to me.

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/authentication/ApiTokenResource.java [hunk 2]
```diff
@@ -166,18 +160,26 @@ public final Response deleteApiToken(@Context final HttpServletRequest request,
     @NoCache
     @Consumes(MediaType.APPLICATION_JSON)
     @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
-    public final Response issueApiToken(@Context final HttpServletRequest request,
-                                   @Context final HttpServletResponse response,
-                                   final ApiTokenForm formData) {
+    public final Response issueApiToken(
```
Comment (erickgonzalez) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/authentication/ApiTokenResource.java:163]: update doc of the endpoint with new params of the form

**Review by erickgonzalez — CHANGES_REQUESTED** [review]

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/authentication/ApiTokenResource.java (hunk 2 above)
Comment (freddyucv) [reply dotCMS/src/main/java/com/dotcms/rest/api/v1/authentication/ApiTokenResource.java:166]: done https://github.com/dotCMS/core/pull/19698/commits/047eca133554e37b9db8a424b1bf434249c6826a

**Review by erickgonzalez — APPROVED** [review]

**Pull request merged**